src/kingdom/Mammal.java

```diff
@@ -0,0 +1,53 @@
+package kingdom;
+
+public class Mammal extends AbstractAnimal {
+
+    private String name;
+    private int year;
+
+    public Mammal(String name, int year) {
+        this.name = name;
+        this.year = year;
+    }
```

Comment:
It seems that `name` and `year` are common among all subclasses of `AbstractAnimal` - you could place them (and the corresponding constructor, which you can still use via the `super` keyword as will be shown below) there instead.

_(Using an inherited constructor from `AbstractAnimal`):_
```
public Mammal (String name, int year) {
    super(name, year);
}
```